cmd/minishift/cmd/delete.go

```diff
@@ -111,6 +121,59 @@ func clearCache() {
 	}
 }
 
+// Remove the current cluster's entries from global kubeconfig file
+func clearKubeConfig(hostIP string) error {
+	kubeConfigPath, err := oc.GetGlobalKubeConfigPath()
+	if err != nil {
+		return err
+	}
+	kubeConfig, err := clientcmd.LoadFromFile(kubeConfigPath)
+	if err != nil {
+		return err
+	}
+	cleanKubeConfig, err := removeEntriesForCluster(hostIP, kubeConfig)
+	if err != nil {
+		return err
+	}
+	return clientcmd.WriteToFile(*cleanKubeConfig, kubeConfigPath)
+}
+
+func removeEntriesForCluster(hostIP string, kubeConfig *clientcmdapi.Config) (*clientcmdapi.Config, error) {
```

Comment:
I am not able to write the unit test for this function. any ideas @praveenkumar

Reply:
@praveenkumar I tried that approach, but we'd need two different kubeconfig files, one _dirty_ and one _clean_ to compare that the o/p of the function is expected. Now the structs inside the config struct has this `LocationOforigin` field which contains the fullpath of the kubeconfig file, which will be different. So i could not make this work.

Reply:
@praveenkumar Went with renaming the files, it works now!